From eb604d6a970ab48591d0a40b3191722afcfe3a09 Mon Sep 17 00:00:00 2001
From: Piyush Mehta <piyush.mehta@xilinx.com>
Date: Wed, 21 Oct 2020 14:49:45 +0530
Subject: [PATCH 1721/1852] usb: dwc3: otg: Enable OTG support in dwc3-core
 driver

commit 840739963767dc31c7ee347d6f94cb54efd32cd8 from
https://github.com/Xilinx/linux-xlnx.git

The Linux v5.4 Kernel does not support OTG feature in the DWC3 USB driver.
It has some checks to set the mode to peripheral when the dr_mode is set
OTG in the device tree. To support the OTG mode for Xilinx platforms, the
check provided by the mainline kernel needs to be removed. This patch
removes the check-in dwc3 driver to make OTG work for Xilinx platforms.

Signed-off-by: Piyush Mehta <piyush.mehta@xilinx.com>
Acked-by: Manish Narani <manish.narani@xilinx.com>
State: not-upstreamable
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/usb/dwc3/core.c | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/drivers/usb/dwc3/core.c b/drivers/usb/dwc3/core.c
index 963707b87990..de0ac61b050d 100644
--- a/drivers/usb/dwc3/core.c
+++ b/drivers/usb/dwc3/core.c
@@ -79,15 +79,6 @@ static int dwc3_get_dr_mode(struct dwc3 *dwc)
 			mode = USB_DR_MODE_HOST;
 		else if (IS_ENABLED(CONFIG_USB_DWC3_GADGET))
 			mode = USB_DR_MODE_PERIPHERAL;
-
-		/*
-		 * DWC_usb31 and DWC_usb3 v3.30a and higher do not support OTG
-		 * mode. If the controller supports DRD but the dr_mode is not
-		 * specified or set to OTG, then set the mode to peripheral.
-		 */
-		if (mode == USB_DR_MODE_OTG &&
-		    dwc->revision >= DWC3_REVISION_330A)
-			mode = USB_DR_MODE_PERIPHERAL;
 	}
 
 	if (mode != dwc->dr_mode) {
-- 
2.31.1

